### PR TITLE
Add logging of GovDelivery response to mailer

### LINKED ISSUE #13627: Log unsent or bounced emails
## Description
Check response from Granicus (through ActionMailer gem) to ensure ability to log and monitor unsent and bounced emails. Log and send alert to Sentry to monitor unsent (and bounced, if possible) Veteran and Representative emails. 

## Out of scope
- Tracking email
- Implement job if email hasn't sent
- Notify hearing coordinator that email didn’t go through

## Background/context/resources
Though we have a current/manual fallback of Hearing Coordinators monitoring an inbox, we need a long-term, scalable solution to log and monitor issues with emails.

### PR DESCRIPTION
**Pull request by ferristseng**
Resolves #13627

### Description

  * Updates `SendEmail` to use `deliver_now!` as opposed to `deliver_now`
  * Updates test expectations
  * Adds `return_response` to GovDelivery config settings, and unsets `config.action_mailer.raise_delivery_errors` because `deliver_now!` ignores that flag 
    * `return_response` isn't well documented, but the behavior is described here: https://stackoverflow.com/questions/6374648/actionmailer-smtp-server-response
  * Refactor: Use `"hearings"` app name for Datadog metrics related to virtual hearings.
  * Remove some unused metrics tracking

*I tested this in UAT partially by explicitly setting `return_response`. We will need to verify that the flag is set after deploying`*

### CODE REVIEW

File: app/jobs/virtual_hearings/send_email.rb [hunk 1]
```diff
@@ -10,17 +10,17 @@ def initialize(virtual_hearing:, type:)
...
     end
 
     if should_judge_receive_email?
-      send_email(:judge)
+      send_email(judge_recipient)
       virtual_hearing.judge_email_sent = true
     end
 
     if !virtual_hearing.representative_email.nil? && !virtual_hearing.representative_email_sent
-      send_email(:representative)
```
Comment (ferristseng) [inline app/jobs/virtual_hearings/send_email.rb]: Upon looking at this, I realized that this could just be replaced by a function.

File: config/environments/production.rb [hunk 2]
```diff
@@ -74,12 +74,13 @@
...
     auth_token: ENV["GOVDELIVERY_TOKEN"],
-    api_root: "https://#{ENV["GOVDELIVERY_SERVER"]}"
+    api_root: "https://#{ENV["GOVDELIVERY_SERVER"]}",
+    return_response: true
   }
 
   # Ignore bad email addresses and do not raise email delivery errors.
   # Set this to true and configure the email server for immediate delivery to raise delivery errors.
-  config.action_mailer.raise_delivery_errors = true
+  # config.action_mailer.raise_delivery_errors = true
```
Comment (ferristseng) [inline config/environments/production.rb]: I'm unsetting this because `deliver_now!` ignores this flag. Since virtual hearings is the first time we're sending emails, this is no longer being used.

File: app/jobs/virtual_hearings/send_email.rb [hunk 3]
```diff
@@ -29,51 +29,90 @@ def call
...
+
+    if msg.is_a?(GovDelivery::TMS::EmailMessage)
+      response = msg.response
+      response_external_url = response.body.dig("_links", "self")
+
+      Rails.logger.info(
+        "GovDelivery returned (code: #{response.status}) (external url: #{response_external_url})"
       )
-    }
+    end
```
Comment (ferristseng) [inline app/jobs/virtual_hearings/send_email.rb]: Tested this in UAT with the following code:

```
me = MailRecipient.new(
  name: "Ferris",
  email: "ferris@navapbc.com",
  title: MailRecipient::RECIPIENT_TITLES[:veteran]
)

email = VirtualHearingMailer.confirmation(
  mail_recipient: me,
  virtual_hearing: VirtualHearing.last
)

email.delivery_method.settings[:return_response] = true

msg = email.deliver_now!

if msg.is_a?(GovDelivery::TMS::EmailMessage)
  response = msg.response
  response_external_url = response.body.dig("_links", "self")

  Rails.logger.info(
    "GovDelivery returned (code: #{response.status}) (external url: #{response_external_url})"
  )
end
```

File: spec/jobs/virtual_hearings/send_email_spec.rb [hunk 4]
```diff
@@ -16,21 +16,24 @@
...
+    VirtualHearings::SendEmail.new(virtual_hearing: virtual_hearing, type: email_type)
   end
-  let(:send_email_job) { VirtualHearings::SendEmail.new(virtual_hearing: virtual_hearing, type: email_type) }
-  let(:virtual_hearing_mailer) { double(VirtualHearingMailer) }
 
   describe ".call" do
     before do
-      allow(VirtualHearings::SendEmail).to receive(:new).and_return(send_email_job)
-      allow(send_email_job).to receive(:mail_recipients).and_return(mail_recipients)
-      allow(virtual_hearing_mailer).to receive(:deliver_now)
```
Comment (ferristseng) [inline spec/jobs/virtual_hearings/send_email_spec.rb]: Removed this because `deliver_now!` / `deliver_now` is a actually called on an instance of `Mail::Message`. It also might be testing a behavior that's too aware of how things work internally

File: app/controllers/appeals_controller.rb [hunk 5]
```diff
@@ -148,16 +146,6 @@ def render_search_results_as_json(result)
     end
   end
 
-  def log_hearings_request
-    # Log requests to this endpoint to try to investigate cause addressed by this rollback:
-    # https://github.com/department-of-veterans-affairs/caseflow/pull/9271
```
Comment (ferristseng) [inline app/controllers/appeals_controller.rb]: Unrelated change: Looked into the issue this is supposed to be gathering metrics for, and it's over a year old, and merged. I think it's probably fine to remove this.

File: app/jobs/virtual_hearings/delete_conferences_job.rb [hunk 6]
```diff
@@ -26,7 +24,7 @@ def perform
 
   def datadog_metric_info
     {
-      app_name: APP_NAME,
```
Comment (ferristseng) [inline app/jobs/virtual_hearings/delete_conferences_job.rb]: `app_name` is used as a tag, so I think setting it to `hearings` makes more sense

File: app/jobs/virtual_hearings/send_email.rb (hunk 3 above)
Comment (rubaiyat22) [reply app/jobs/virtual_hearings/send_email.rb]: send myself an email in UAT and it worked! 🎉
